budgeteer-rest/src/main/java/de/adesso/budgeteer/rest/invoice/InvoiceController.java

```diff
@@ -0,0 +1,79 @@
+package de.adesso.budgeteer.rest.invoice;
+
+import de.adesso.budgeteer.core.invoice.port.in.*;
+import de.adesso.budgeteer.rest.invoice.model.CreateInvoiceModel;
+import de.adesso.budgeteer.rest.invoice.model.InvoiceModel;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/invoices")
+public class InvoiceController {
+    private final GetInvoiceByIdUseCase getInvoiceByIdUseCase;
+    private final GetInvoicesInContractUseCase getInvoicesInContractUseCase;
+    private final GetInvoicesInProjectUseCase getInvoicesInProjectUseCase;
+    private final DeleteInvoiceByIdUseCase deleteInvoiceByIdUseCase;
+    private final CreateInvoiceUseCase createInvoiceUseCase;
+    private final UpdateInvoiceUseCase updateInvoiceUseCase;
+    private final InvoiceModelMapper invoiceModelMapper;
+
+    @GetMapping("/{invoiceId}")
+    public Optional<InvoiceModel> getInvoice(@PathVariable long invoiceId) {
+        return getInvoiceByIdUseCase.getInvoiceById(invoiceId).map(invoiceModelMapper::toModel);
+    }
+
+    @GetMapping
+    public List<InvoiceModel> getInvoicesInContract(@RequestParam("contractId") long contractId) {
+        return invoiceModelMapper.toModel(getInvoicesInContractUseCase.getInvoicesInContract(contractId));
+    }
+
+    @GetMapping
+    public List<InvoiceModel> getInvoicesInProject(@RequestParam("projectId") long projectId) {
```

Comment:
It looks like we can't use the same mapping, but with different request parameters. So we either have to take either your first approach of two completely different mappings, or the instead merge them to one method which takes two optional request parameters, that checks which one was set, throws an error if both or none are, and returns the correct list otherwise.